src/main/java/me/karakelley/http/controllers/Application.java

```diff
@@ -1,28 +1,38 @@
 package me.karakelley.http.controllers;
 
-import me.karakelley.http.PublicDirectory;
+import me.karakelley.http.FileSystem.PublicDirectory;
 import me.karakelley.http.Request;
 import me.karakelley.http.Response;
+import me.karakelley.http.ContentGeneration.HtmlGenerator;
 
 public class Application implements Controller {
   private final Router router = new Router();
+  private PublicDirectory publicDirectory = null;
 
   public Application() {
     router.route("/", new HelloWorldController());
     setupCommonRoutes();
   }
 
   public Application(PublicDirectory publicDirectory) {
-    router.route("/", new StaticFilesController());
+    this.publicDirectory = publicDirectory;
+    router.route("/", new StaticFilesController(publicDirectory, new HtmlGenerator()));
     setupCommonRoutes();
   }
 
   @Override
   public Response respond(Request request) {
+    addRouteForRequestedResource(request);
     return router.respond(request);
   }
 
   private void setupCommonRoutes() {
     router.route("/redirectme", new RedirectController());
   }
+
+  private void addRouteForRequestedResource(Request request) {
+    if (publicDirectory != null && publicDirectory.resourceExists(request.getPath())) {
+      router.route(request.getPath(), new StaticFilesController(publicDirectory, new HtmlGenerator()));
```

Comment:
I still think this is a really strange solution to this problem. Can you just call into the `StaticFilesController` and let it return the files which correspond to the path?

Reply:
I don't quite understand what you are suggesting is strange about the solution. Validating the resource exists and if there's a directory or how I'm adding the route?